src/MSBuild/XMake.cs

```diff
@@ -2514,10 +2515,46 @@ private static bool GatherAutoResponseFileSwitchesFromFullPath(string autoRespon
             return invokeBuild;
         }
 
-        private static bool ProcessTerminalLoggerConfiguration(CommandLineSwitches commandLineSwitches)
+        private static bool ProcessTerminalLoggerConfiguration(CommandLineSwitches commandLineSwitches, out string aggregatedParameters)
         {
             string terminalloggerArg;
 
+            string[] terminalLoggerParameters = commandLineSwitches[CommandLineSwitches.ParameterizedSwitch.TerminalLoggerParameters];
+            aggregatedParameters = terminalLoggerParameters?.Length > 0 ?
+                AggregateParameters(string.Empty, terminalLoggerParameters) :
+                string.Empty;
+
+            // Find default configuration so it is part of telemetry even when default is not used.
+            // Default can be stored in /tlp:default=true|false|on|off|auto
+            string defaultValue = null;
+            foreach (string parameter in aggregatedParameters.Split(MSBuildConstants.SemicolonChar))
+            {
+                if (string.IsNullOrWhiteSpace(parameter))
+                {
+                    continue;
+                }
+
+                string[] parameterAndValue = parameter.Split(MSBuildConstants.EqualsChar);
+                if (parameterAndValue[0].Equals("DEFAULT", StringComparison.InvariantCultureIgnoreCase) && parameterAndValue.Length > 1)
+                {
+                    defaultValue = parameterAndValue[1];
+                }
+            }
+
+            if (defaultValue == null)
+            {
+                defaultValue = bool.FalseString;
+                KnownTelemetry.LoggingConfigurationTelemetry.TerminalLoggerDefault = bool.FalseString;
+                KnownTelemetry.LoggingConfigurationTelemetry.TerminalLoggerDefaultSource = "msbuild";
+            }
+            else
+            {
+                // Lets check DOTNET CLI env var
+                string dotnetCliEnvVar = Environment.GetEnvironmentVariable("DOTNET_CLI_BUILD_TERMINAL_LOGGER");
+                KnownTelemetry.LoggingConfigurationTelemetry.TerminalLoggerDefault = defaultValue;
+                KnownTelemetry.LoggingConfigurationTelemetry.TerminalLoggerDefaultSource = string.IsNullOrEmpty(dotnetCliEnvVar) ? "sdk" : "DOTNET_CLI_BUILD_TERMINAL_LOGGER";
```

Comment:
line 2532 used `string.IsNullOrWhiteSpace()` consider homogeneity unless there are specific reasons to use both.

Reply:
Fixed